server/backends/blobstore/aws/aws.go

```diff
@@ -254,3 +256,38 @@ func (a *AwsS3BlobStore) BlobExists(ctx context.Context, blobName string) (bool,
 
 	return true, nil
 }
+
+func (a *AwsS3BlobStore) Writer(ctx context.Context, blobName string) (interfaces.CommittedWriteCloser, error) {
+	// Open a pipe.
+	pr, pw := io.Pipe()
+
+	errch := make(chan error, 1)
+
+	ctx, cancel := context.WithCancel(ctx)
+
+	// Upload from pr in a separate Go routine.
+	go func() {
+		defer pr.Close()
+		_, err := a.uploader.UploadWithContext(
+			ctx,
+			&s3manager.UploadInput{
+				Bucket: a.bucket,
+				Key:    aws.String(util.BlobPath(blobName)),
+				Body:   pr,
+			},
+		)
+		errch <- err
+		close(errch)
+	}()
+
+	committer := &util.CustomCommitter{CommitOp: func() error {
+		return (&util.AsyncCloser{Closer: pw, ErrorChannel: errch}).Close()
```

Comment:
Since the `AsyncCloser` is only being used in these `committer` funcs now, and the `Close()` logic is relatively short (*), wdyt about inlining the logic here and removing the AsyncCloser class?

(*) the `Close()` logic could also be a little shorter:

```
	if err := pw.Close(); err != nil {
    // This next if {} block can be removed
    // since errch is buffered
		if err := <-a.ErrorChannel; err != nil {
			return err
		}
		return err
	}
	return <-errch
```